org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/Activator.java

```diff
@@ -62,5 +64,14 @@ private static void setDefault(Activator activator) {
 	public static ImageDescriptor getImageDescriptor(String path) {
 		return imageDescriptorFromPlugin(PLUGIN_ID, path);
 	}
+	
+	/**
+	 * TODO: remove this workaround as soon as a clear solution is
+	 * provided to make the UiInjectorProvider registry aware.
+	 */
+	@Override
```

Comment:
@szarnekow @miklossy what is this needed for?

Reply:
The PR contains actually two commits. The first commit contains the refactoring, but that lead to some failing XtextGrammarQuickfixProviderTest test cases due to the fact that the UiInjectorProvider has not been registry aware. If that problem has already been solved, that workaround is not necessary any more and the second commit can be removed from the PR.